src/main/java/com/cs3365/rpgdapi/weapons/WeaponsService.java

```diff
@@ -5,4 +5,5 @@
 public interface WeaponsService {
     UUID createWeapon(Weapon requestBody) throws WeaponsException;
     WeaponEntity findWeapon(UUID identity) throws WeaponsException;
+    void updateWeapon(WeaponEntity weapon) throws WeaponsException;
```

Comment:
This needs to have the `WeaponEntity` return type, and update the signature to include `UUID identifier`. You will need to pass the identifier to the service to use it to fetch what's in the database before updating it.

This may not make much sense now, but when the codebase increases in size and becomes more convoluted this is an important thing to consider.